### PR TITLE
Fix `push-images` workflow

### PR DESCRIPTION
**Pull request by harupy**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Fixes the `push-images` workflow

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: .github/workflows/push-images.yml [hunk 1]
```diff
@@ -2,7 +2,9 @@ name: Push-Images
 
 on:
   release:
-    types: [published]
+    types:
+      - published
+      - edited
```
Comment (harupy) [inline .github/workflows/push-images.yml:7]: This allows us to rerun the job by editing a release note.

File: README.rst [hunk 2]
```diff
@@ -63,6 +63,8 @@ Nightly Job Statuses
     :target: https://github.com/mlflow/mlflow/actions?query=workflow%3ATest%2Brequirements+event%3Aschedule
 .. |stale| image:: https://img.shields.io/github/actions/workflow/status/mlflow/mlflow/stale.yml?branch=master&event=schedule&label=stale&logo=github&style=for-the-badge
     :target: https://github.com/mlflow/mlflow/actions?query=workflow%3AStale+event%3Aschedule
+.. |push-images| image:: https://img.shields.io/github/actions/workflow/status/mlflow/mlflow/push-images.yml?event=release&label=push-images&logo=github&style=for-the-badge
+    :target: https://github.com/mlflow/mlflow/actions/workflows/push-images.yml?query=event%3Arelease
```
Comment (harupy) [inline README.rst:67]: image

File: .github/workflows/push-images.yml [hunk 3]
```diff
@@ -69,7 +71,9 @@ jobs:
 
       - name: Push Model Server Image
         run: |
-          for tag in ${{ steps.modelmeta.outputs.tags }}; do
+          set -x
+
+          for tag in ${{ join(steps.modelmeta.outputs.tags, ' ') }}; do
```
Comment (harupy) [inline .github/workflows/push-images.yml:76]: `${{ steps.modelmeta.outputs.tags }}` results in a multi-line string, hence join:

https://github.com/mlflow/mlflow/actions/runs/4412341741/jobs/7731711749

```
  for tag in ghcr.io/mlflow/model-server:v2.2.2
  ghcr.io/mlflow/model-server:latest; do
    docker tag model-server:latest $tag
    docker push $tag
  done
```

**Review by WeichenXu123 — APPROVED** [review]
LGTM!

**Pull request merged**